test/e2e/e2e_test.go

```diff
@@ -170,6 +170,53 @@ func checkConnectivityPingToHost(f *framework.Framework, nodeName, podName, host
 	return err
 }
 
+// Place the workload on the specified node to check pod default routes go thru .3 address
+func checkPodRoutes(f *framework.Framework, nodeName string, podName string, exGw bool) error {
+	contName := fmt.Sprintf("%s-container", podName)
+	command := []string{"/bin/sh", "-c"}
+	args := []string{fmt.Sprintf("sleep 300")}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    contName,
+					Image:   framework.AgnHostImage,
+					Command: command,
+					Args:    args,
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	_, err := podClient.Create(pod)
+	if err != nil {
+		return err
+	}
+
+	// Wait for pod network setup to be almost ready
+	wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
```

Comment:
if we don't check for an error here we don't know if it failed the check

Reply:
this should be 
```suggestion
	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
```

and the chek if err != nil

full disclaimer, the external gateway annotations on pods does not work, it seems the pod never gets the annotation
https://github.com/ovn-org/ovn-kubernetes/pull/1561#discussion_r485782645

Reply:
we talked offline, and seems pod does not receive annotations, so we need to check for them on the pod